pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -106,13 +107,19 @@ func (r *DeploymentReconciler) checkDeploymentExist(client client.Client) (const
 		return constants.CheckResultUnknown, nil, err
 	}
 	//existed, check equivalence
-	if semanticDeploymentEquals(r.Deployment, existingDeployment) {
+	if semanticDeploymentEquals(*r.Deployment, *existingDeployment) {
 		return constants.CheckResultExisted, existingDeployment, nil
 	}
+
+	diff, err := kmp.SafeDiff(r.Deployment.Spec, existingDeployment.Spec)
+	log.Info("Deployment Updated", "Diff", diff, "err", err)
+
 	return constants.CheckResultUpdate, existingDeployment, nil
 }
 
-func semanticDeploymentEquals(desired, existing *appsv1.Deployment) bool {
+func semanticDeploymentEquals(desired, existing appsv1.Deployment) bool {
+	//for HPA scaling, we should ignore Replicas of Deployment
+	*desired.Spec.Replicas = *existing.Spec.Replicas
```

Comment:
probably not a good idea to mutate the field in an equal function

Reply:
according to https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#migrating-deployments-and-statefulsets-to-horizontal-autoscaling we should not set replica field if HPA is used.

Reply:
good idea, now we removed `Replicas` from the `Deployment Rconciler`, and use safe diff + ignoreFields for implementation.